packages/suite/src/actions/wallet/send/sendFormEthereumActions.ts

```diff
@@ -109,34 +109,45 @@ export const composeTransaction = (
 
     const { output, tokenInfo, decimals } = composeOutputs;
     const { availableBalance } = account;
-    const { address } = formValues.outputs[0];
+    const { address, amount } = formValues.outputs[0];
 
-    // additional calculation for gasLimit based on data size
     let customFeeLimit: string | undefined;
-    if (typeof formValues.ethereumDataHex === 'string' && formValues.ethereumDataHex.length > 0) {
-        const response = await TrezorConnect.blockchainEstimateFee({
-            coin: account.symbol,
-            request: {
-                blocks: [2],
-                specific: {
-                    from: account.descriptor,
-                    to: address || account.descriptor,
-                    data: formValues.ethereumDataHex,
-                },
-            },
-        });
-
-        if (response.success) {
-            customFeeLimit = response.payload.levels[0].feeLimit;
-        }
-    }
-
     // set gasLimit based on ERC20 transfer
     if (tokenInfo) {
         customFeeLimit = ERC20_GAS_LIMIT;
     }
 
-    const predefinedLevels = feeInfo.levels.filter(l => l.label !== 'custom');
+    // gasLimit calculation based on address, amount and data size
+    // amount in essential for a proper calculation of gasLimit (via blockbook/geth)
+    const estimatedFee = await TrezorConnect.blockchainEstimateFee({
```

Comment:
just asking - is it needed to have the code duplicated in send and coinmarket?

Reply:
nope, they should be shared at some point
there are slight differences, and both functionalities was developed at the same time +/-

in ideal world there should be only 1 action for that, we will get there :)